add task solution

Ivakhovd opened the pull request.

mateacademy-ai-mentor commented on src/styles/card.scss:
```diff
@@ -0,0 +1,112 @@
+  height: 408px;
+  border-radius: 5px;
+  border: 1px solid #f3f3f3;
+  background: $background;
```
The variable `$background` is used here, but it's not defined within this file. Ensure that it is defined in an imported file or within this file to avoid compilation errors.

mateacademy-ai-mentor commented on src/styles/card.scss:
```diff
@@ -0,0 +1,112 @@
+  }
+
+  &__text {
+    padding-left: $padding-left;
```
The variable `$padding-left` is used here, but it's not defined within this file. Ensure that it is defined in an imported file or within this file to avoid compilation errors.

mateacademy-ai-mentor commented on src/styles/card.scss:
```diff
@@ -0,0 +1,112 @@
+    font-size: 12px;
+    font-style: normal;
+    font-weight: 500;
+    line-height: $line-height;
```
The variable `$line-height` is used here, but it's not defined within this file. Ensure that it is defined in an imported file or within this file to avoid compilation errors.

mateacademy-ai-mentor commented on src/styles/card.scss:
```diff
@@ -0,0 +1,112 @@
+    font-style: normal;
+    font-weight: 500;
+    line-height: $line-height;
+    color: $color-text;
```
The variable `$color-text` is used here, but it's not defined within this file. Ensure that it is defined in an imported file or within this file to avoid compilation errors.

mateacademy-ai-mentor commented on src/styles/card.scss:
```diff
@@ -0,0 +1,112 @@
+  font-size: 10px;
+  font-weight: 400;
+  line-height: 14px;
+  color: $color-text-next;
```
The variable `$color-text-next` is used here, but it's not defined within this file. Ensure that it is defined in an imported file or within this file to avoid compilation errors.

mateacademy-ai-mentor commented on src/styles/card.scss:
```diff
@@ -0,0 +1,112 @@
+}
+
+.card__reviews {
+  padding-right: $padding-left;
```
The variable `$padding-left` is used here, but it's not defined within this file. Ensure that it is defined in an imported file or within this file to avoid compilation errors.

mateacademy-ai-mentor commented on src/styles/card.scss:
```diff
@@ -0,0 +1,112 @@
+
+.card__reviews {
+  padding-right: $padding-left;
+  color: $color-text;
```
The variable `$color-text` is used here, but it's not defined within this file. Ensure that it is defined in an imported file or within this file to avoid compilation errors.

mateacademy-ai-mentor commented on src/styles/card.scss:
```diff
@@ -0,0 +1,112 @@
+}
+
+.text--four {
+  color: $color-text;
```
The variable `$color-text` is used here, but it's not defined within this file. Ensure that it is defined in an imported file or within this file to avoid compilation errors.

mateacademy-ai-mentor commented on src/styles/card.scss:
```diff
@@ -0,0 +1,112 @@
+  text-align: right;
+  font-size: 16px;
+  font-weight: 700;
+  line-height: $line-height;
```
The variable `$line-height` is used here, but it's not defined within this file. Ensure that it is defined in an imported file or within this file to avoid compilation errors.

mateacademy-ai-mentor commented on src/styles/card.scss:
```diff
@@ -0,0 +1,112 @@
+}
+
+.text--three {
+  color: $color-text-next;
```
The variable `$color-text-next` is used here, but it's not defined within this file. Ensure that it is defined in an imported file or within this file to avoid compilation errors.

mateacademy-ai-mentor commented on src/styles/card.scss:
```diff
@@ -0,0 +1,112 @@
+.text--three {
+  color: $color-text-next;
+  font-size: 12px;
+  line-height: $line-height;
```
The variable `$line-height` is used here, but it's not defined within this file. Ensure that it is defined in an imported file or within this file to avoid compilation errors.

mateacademy-ai-mentor commented on src/styles/card.scss:
```diff
@@ -0,0 +1,112 @@
+  color: $color-text-next;
+  font-size: 12px;
+  line-height: $line-height;
+  padding-left: $padding-left;
```
The variable `$padding-left` is used here, but it's not defined within this file. Ensure that it is defined in an imported file or within this file to avoid compilation errors.

mateacademy-ai-mentor commented on src/styles/card.scss:
```diff
@@ -0,0 +1,112 @@
+  font-weight: 700;
+  width: 166px;
+  height: 40px;
+  background-color: $color-border;
```
The variable `$color-border` is used here, but it's not defined within this file. Ensure that it is defined in an imported file or within this file to avoid compilation errors.

mateacademy-ai-mentor commented on src/styles/card.scss:
```diff
@@ -0,0 +1,112 @@
+.card__button:hover {
+  cursor: pointer;
+  border-radius: 5px;
+  border: 1px solid $color-border;
```
The variable `$color-border` is used here, but it's not defined within this file. Ensure that it is defined in an imported file or within this file to avoid compilation errors.

mateacademy-ai-mentor commented on src/styles/card.scss:
```diff
@@ -0,0 +1,112 @@
+  cursor: pointer;
+  border-radius: 5px;
+  border: 1px solid $color-border;
+  background: $background;
```
The variable `$background` is used here, but it's not defined within this file. Ensure that it is defined in an imported file or within this file to avoid compilation errors.

mateacademy-ai-mentor commented on src/styles/card.scss:
```diff
@@ -0,0 +1,112 @@
+  border-radius: 5px;
+  border: 1px solid $color-border;
+  background: $background;
+  color: $color-border;
```
The variable `$color-border` is used here, but it's not defined within this file. Ensure that it is defined in an imported file or within this file to avoid compilation errors.

mateacademy-ai-mentor commented on src/styles/stars.scss:
```diff
@@ -0,0 +1,23 @@
+    display: flex;
+    gap: 2px;
+    align-items: center;
+    padding-left: $padding-left;
```
The variable `$padding-left` is used here, but it's not defined within this file. Ensure that it is defined in an imported file or within this file to avoid compilation errors.

mateacademy-ai-mentor reviewed (changes requested):
It looks like there are several SCSS variables being used that are not defined within the files or imported from another file. This could lead to compilation errors and affect the functionality of your code. Please ensure that all variables such as `$padding-left`, `$color-border`, `$background`, `$line-height`, `$color-text`, and `$color-text-next` are properly defined in your `variables.scss` file or imported correctly. Once these issues are resolved, your implementation should be ready to go! Keep up the great work and attention to detail! 😊

✨ Please rate this review in the Platform's notification center. I will be grateful for your feedback! ✨

🔄 Fix the issues, commit the changes, and then re-request my review using the `Re-request review` button in the Reviewers tab.

volodymyr-soltys97 commented on readme.md:
```diff
@@ -20,8 +20,8 @@ Create a pages with product card using `flexbox`, `BEM` and `SCSS` based on [thi
 
-- [DEMO LINK](https://<your_account>.github.io/layout_product-cards/)
-- [TEST REPORT LINK](https://<your_account>.github.io/layout_product-cards/report/html_report/)
+- [DEMO LINK](https://Ivakhovd.github.io/layout_product-cards/)
+- [TEST REPORT LINK](https://Ivakhovd.github.io/layout_product-cards/report/html_report/)
```
Add these links to the PR description 
<img width="927" alt="image" src="https://github.com/user-attachments/assets/cdd11441-0d1b-41a0-9c61-cd20201ab539">

volodymyr-soltys97 commented on src/index.html:
```diff
@@ -11,8 +11,55 @@
+        class="card__img"
+        alt="picture"
+      />
+      <p class="card__text">
```
```suggestion
      <p class="card__title">
```

volodymyr-soltys97 commented on src/index.html:
```diff
@@ -11,8 +11,55 @@
+        <br />
+        5K Monoblock (MNED2UA/A)
+      </p>
+      <p class="text text--one">Product code: 195434</p>
```
```suggestion
      <p class="card__subtitle">Product code: 195434</p>
```

volodymyr-soltys97 commented on src/index.html:
```diff
@@ -11,8 +11,55 @@
+        <p class="card__reviews">Reviews: 5</p>
+      </div>
+      <div class="card__price">
+        <p class="card__price text text--three">Price:</p>
+        <p class="card__price text text--four">$2,199</p>
```
```suggestion
        <p class="card__price--label">Price:</p>
        <p class="card__price--value">$2,199</p>
```

volodymyr-soltys97 commented on src/styles/card.scss:
```diff
@@ -0,0 +1,112 @@
+  flex-direction: column;
+  box-sizing: border-box;
+  width: 200px;
+  height: 408px;
```
Don't need to write fixed height for the card

volodymyr-soltys97 replied:
```suggestion
```

volodymyr-soltys97 commented on src/styles/card.scss:
```diff
@@ -0,0 +1,112 @@
+  align-items: center;
+  justify-content: center;
+  box-sizing: border-box;
+}
+
+.card__price {
```
Also, you need to use scss nesting, fix it everywhere

volodymyr-soltys97 reviewed (changes requested):
Good job 👍 
Let's improve your code
This part of the task is incomplete, you need to use styles from the previous task 
<img width="621" alt="image" src="https://github.com/user-attachments/assets/02cac2a2-b634-4e1a-815c-387c6c579ba3">
<img width="1512" alt="image" src="https://github.com/user-attachments/assets/b90dee66-b26a-4a89-9143-1a5c4fb67495">

Ivakhovd commented:
- [DEMO LINK](https://Ivakhovd.github.io/layout_product-cards/)
- [TEST REPORT LINK](https://Ivakhovd.github.io/layout_product-cards/report/html_report/)

volodymyr-soltys97 commented on src/styles/stars.scss:
```diff
@@ -0,0 +1,23 @@
+.stars {
+  &--4 {
+    display: flex;
+    gap: 2px;
+    align-items: center;
+    padding-left: $padding-left;
+    width: 96px;
+    height: 16px;
+  }
```
You need to implement the modifier logic, use styles for stars from the previous task

volodymyr-soltys97 reviewed (changes requested):
Hi, this comment still not fixed from the previous review 
<img width="933" alt="image" src="https://github.com/user-attachments/assets/5d749410-0856-47cf-934c-99cd7bb60448">

And this 
<img width="890" alt="image" src="https://github.com/user-attachments/assets/3b655274-bbd8-4168-8795-052ea1e3bae8">

Ivakhovd commented:
@volodymyr-soltys97  I really don't understand what the problem is, I use styles as in the previous task with stars (the only difference I corrected is that I put margin-right instead of padding-right), I also added a demo link and a test report link to the PR comment. If something is wrong, please specify the error, what and how to fix it, because as long as the mentor accepts the task, I will play with him for a month instead of moving on

volodymyr-soltys97 reviewed (changes requested):
Hi, the comments from the previous review still fixed, if you need some help feel free in the fe_chat. When we change the number 4 to 2 in the name of the modifier class, for example, there should be 2 active stars on the page, but now everything breaks

volodymyr-soltys97 reviewed (changes requested):
Hi, you need to fix demo link 
<img width="308" alt="image" src="https://github.com/user-attachments/assets/2859e24c-2aa6-44e5-ba21-6e2b110a45ec">

volodymyr-soltys97 reviewed (changes requested):
Hi, I answered your in the fe chat how to fix it 
<img width="1512" alt="image" src="https://github.com/user-attachments/assets/7efe9329-105c-4d04-b89b-43c760787140">

volodymyr-soltys97 reviewed (changes requested):
Hi, there is no commit from the previous review, also, you need to make a deploy again after your changes and check it after on the demo
<img width="1189" alt="image" src="https://github.com/user-attachments/assets/d0ed2fc9-a5d5-49aa-9685-0c4df4758ee6"> 
<img width="972" alt="image" src="https://github.com/user-attachments/assets/e337d123-654c-41f7-9064-9ae81d31db93">
